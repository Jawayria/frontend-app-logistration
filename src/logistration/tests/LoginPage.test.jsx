import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';

import LoginPage from '../LoginPage';

const IntlLoginPage = injectIntl(LoginPage);
const mockStore = configureStore();

describe('LoginPage', () => {
  const initialState = {
    logistration: {
      forgotPassword: { status: null },
      loginResult: { success: false, redirectUrl: '' },
      thirdPartyAuthContext: { secondaryProviders: [] },
      response_error: null,
    },
  };

  let props = {};
  let store = {};

  const reduxWrapper = children => (
    <IntlProvider locale="en">
      <Provider store={store}>{children}</Provider>
    </IntlProvider>
  );

  beforeEach(() => {
    store = mockStore(initialState);
    props = {
      loginRequest: jest.fn(),
    };
  });

  it('should match default section snapshot', () => {
    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match forget password alert message snapshot', () => {
    props = {
      ...props,
      forgotPassword: { status: 'complete', email: 'test@example.com' },
    };
    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display login help button', () => {
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.find('button.field-link').text()).toEqual('Need help signing in?');
  });

  it('updates the error state for invalid email', () => {
    const errorState = { email: null, password: null };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('button.submit').simulate('click');
    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('updates the error state for invalid password', () => {
    const errorState = { email: '', password: null };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('input#loginEmail').simulate('change', { target: { value: 'test@example.com', name: 'email' } });
    loginPage.find('button.submit').simulate('click');
    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('should match url after redirection', () => {
    const dasboardUrl = 'http://test.com/testing-dashboard/';
    store = mockStore({
      ...store,
      logistration: {
        ...store.logistration,
        thirdPartyAuthContext: { secondaryProviders: [] },
        loginResult: {
          success: true,
          redirectUrl: dasboardUrl,
        },
      },
    });
    delete window.location;
    window.location = { href: '' };
    renderer.create(reduxWrapper(<IntlLoginPage {...props} />));
    expect(window.location.href).toBe(dasboardUrl);
  });

  it('should call the componentDidMount lifecycle method', () => {
    const spy = jest.spyOn(LoginPage.WrappedComponent.prototype, 'componentDidMount');

    mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(spy).toHaveBeenCalled();
  });

  it('should show error message on 400', () => {
    store = mockStore({
      ...store,
      logistration: {
        ...store.logistration,
        thirdPartyAuthContext: { secondaryProviders: [] },
        loginResult: {
          success: false,
          redirectUrl: '',
        },
        loginError: 'Email or password is incorrect.',
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show error message on 400 on receiving link', () => {
    store = mockStore({
      ...store,
      logistration: {
        ...store.logistration,
        thirdPartyAuthContext: { secondaryProviders: [] },
        loginResult: {
          success: false,
          redirectUrl: '',
        },
        loginError: 'To be on the safe side, you can reset your password <a href="/reset">here</a> before you try again.\n',
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display institution login button', () => {
    store = mockStore({
      ...store,
      logistration: {
        ...store.logistration,
        loginResult: {
          success: true,
          redirectUrl: '',
        },
        thirdPartyAuthContext: {
          secondaryProviders: [
            {
              id: 'saml-test',
              name: 'Test University',
              loginUrl: '/dummy-auth',
              registerUrl: '/dummy_auth',
            },
          ],
        },
      },
    });
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.text().includes('Use my university info')).toBe(true);
  });

  it('should not display institution login button', () => {
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.text().includes('Use my university info')).toBe(false);
  });

  it('should display institution login page', () => {
    store = mockStore({
      ...store,
      logistration: {
        ...store.logistration,
        loginResult: {
          success: true,
          redirectUrl: '',
        },
        thirdPartyAuthContext: {
          secondaryProviders: [
            {
              id: 'saml-test',
              name: 'Test University',
              loginUrl: '/dummy-auth',
              registerUrl: '/dummy_auth',
            },
          ],
        },
      },
    });
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    loginPage.find('button.submit').at(0).simulate('click', { institutionLogin: true });
    expect(loginPage.text().includes('Test University')).toBe(true);
  });
});
