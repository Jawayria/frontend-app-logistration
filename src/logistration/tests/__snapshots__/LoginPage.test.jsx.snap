// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match default section snapshot 1`] = `
<div
  className="d-flex justify-content-center logistration-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account.
        </a>
      </p>
    </div>
    <form
      className="m-0"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Sign In
        </h3>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginEmail"
            >
              Email
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginEmail"
              name="email"
              onChange={[Function]}
              placeholder="username@domain.com"
              style={
                Object {
                  "width": "400px",
                }
              }
              type="email"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="email-invalid-feedback"
            >
              The email address you've provided isn't formatted correctly.
            </strong>
          </div>
        </div>
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group mb-0"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginPassword"
              name="password"
              onChange={[Function]}
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="password-invalid-feedback"
            >
              Please enter your password.
            </strong>
          </div>
        </div>
        <button
          className="mt-0 field-link btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
      </div>
      <button
        className="btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign in
      </button>
    </form>
    <div
      className="section-heading-line mb-4"
    >
      <h4>
        or sign in with
      </h4>
    </div>
    <div
      className="row text-center d-block mb-4"
    >
      <button
        className="btn-social facebook"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook-f fa-w-10 mr-2"
          data-icon="facebook-f"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Facebook
      </button>
      <button
        className="btn-social google"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 mr-2"
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Google
      </button>
      <button
        className="btn-social microsoft"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microsoft fa-w-14 mr-2"
          data-icon="microsoft"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32h214.6v214.6H0V32zm233.4 0H448v214.6H233.4V32zM0 265.4h214.6V480H0V265.4zm233.4 0H448V480H233.4V265.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Microsoft
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage should match forget password alert message snapshot 1`] = `
<div
  className="d-flex justify-content-center logistration-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account.
        </a>
      </p>
    </div>
    <form
      className="m-0"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Sign In
        </h3>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginEmail"
            >
              Email
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginEmail"
              name="email"
              onChange={[Function]}
              placeholder="username@domain.com"
              style={
                Object {
                  "width": "400px",
                }
              }
              type="email"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="email-invalid-feedback"
            >
              The email address you've provided isn't formatted correctly.
            </strong>
          </div>
        </div>
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group mb-0"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginPassword"
              name="password"
              onChange={[Function]}
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="password-invalid-feedback"
            >
              Please enter your password.
            </strong>
          </div>
        </div>
        <button
          className="mt-0 field-link btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
      </div>
      <button
        className="btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign in
      </button>
    </form>
    <div
      className="section-heading-line mb-4"
    >
      <h4>
        or sign in with
      </h4>
    </div>
    <div
      className="row text-center d-block mb-4"
    >
      <button
        className="btn-social facebook"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook-f fa-w-10 mr-2"
          data-icon="facebook-f"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Facebook
      </button>
      <button
        className="btn-social google"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 mr-2"
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Google
      </button>
      <button
        className="btn-social microsoft"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microsoft fa-w-14 mr-2"
          data-icon="microsoft"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32h214.6v214.6H0V32zm233.4 0H448v214.6H233.4V32zM0 265.4h214.6V480H0V265.4zm233.4 0H448V480H233.4V265.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Microsoft
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage should show error message on 400 1`] = `
<div
  className="d-flex justify-content-center logistration-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="fade alert alert-danger show"
      role="alert"
    >
      <div>
        <h4
          style={
            Object {
              "color": "#a0050e",
            }
          }
        >
          <span>
            We couldn't sign you in.
          </span>
        </h4>
        <ul>
          <li>
            Email or password is incorrect.
          </li>
        </ul>
      </div>
    </div>
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account.
        </a>
      </p>
    </div>
    <form
      className="m-0"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Sign In
        </h3>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginEmail"
            >
              Email
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginEmail"
              name="email"
              onChange={[Function]}
              placeholder="username@domain.com"
              style={
                Object {
                  "width": "400px",
                }
              }
              type="email"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="email-invalid-feedback"
            >
              The email address you've provided isn't formatted correctly.
            </strong>
          </div>
        </div>
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group mb-0"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginPassword"
              name="password"
              onChange={[Function]}
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="password-invalid-feedback"
            >
              Please enter your password.
            </strong>
          </div>
        </div>
        <button
          className="mt-0 field-link btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
      </div>
      <button
        className="btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign in
      </button>
    </form>
    <div
      className="section-heading-line mb-4"
    >
      <h4>
        or sign in with
      </h4>
    </div>
    <div
      className="row text-center d-block mb-4"
    >
      <button
        className="btn-social facebook"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook-f fa-w-10 mr-2"
          data-icon="facebook-f"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Facebook
      </button>
      <button
        className="btn-social google"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 mr-2"
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Google
      </button>
      <button
        className="btn-social microsoft"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microsoft fa-w-14 mr-2"
          data-icon="microsoft"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32h214.6v214.6H0V32zm233.4 0H448v214.6H233.4V32zM0 265.4h214.6V480H0V265.4zm233.4 0H448V480H233.4V265.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Microsoft
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage should show error message on 400 on receiving link 1`] = `
<div
  className="d-flex justify-content-center logistration-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="fade alert alert-danger show"
      role="alert"
    >
      <div>
        <h4
          style={
            Object {
              "color": "#a0050e",
            }
          }
        >
          <span>
            We couldn't sign you in.
          </span>
        </h4>
        <ul>
          <li>
            To be on the safe side, you can reset your password 
            <a
              href="/reset"
              onClick={[Function]}
              target="_self"
            >
              here
            </a>
             before you try again.
          </li>
        </ul>
      </div>
    </div>
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account.
        </a>
      </p>
    </div>
    <form
      className="m-0"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Sign In
        </h3>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginEmail"
            >
              Email
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginEmail"
              name="email"
              onChange={[Function]}
              placeholder="username@domain.com"
              style={
                Object {
                  "width": "400px",
                }
              }
              type="email"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="email-invalid-feedback"
            >
              The email address you've provided isn't formatted correctly.
            </strong>
          </div>
        </div>
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group mb-0"
          >
            <label
              className="h6 mr-1"
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="loginPassword"
              name="password"
              onChange={[Function]}
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="password-invalid-feedback"
            >
              Please enter your password.
            </strong>
          </div>
        </div>
        <button
          className="mt-0 field-link btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
      </div>
      <button
        className="btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Sign in
      </button>
    </form>
    <div
      className="section-heading-line mb-4"
    >
      <h4>
        or sign in with
      </h4>
    </div>
    <div
      className="row text-center d-block mb-4"
    >
      <button
        className="btn-social facebook"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook-f fa-w-10 mr-2"
          data-icon="facebook-f"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Facebook
      </button>
      <button
        className="btn-social google"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 mr-2"
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Google
      </button>
      <button
        className="btn-social microsoft"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microsoft fa-w-14 mr-2"
          data-icon="microsoft"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32h214.6v214.6H0V32zm233.4 0H448v214.6H233.4V32zM0 265.4h214.6V480H0V265.4zm233.4 0H448V480H233.4V265.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Microsoft
      </button>
    </div>
  </div>
</div>
`;
